### PR TITLE
Fixes for Python 3

### PR DESCRIPTION
**Pull request by yontalcar**
These commits should fix most of issues tagged ‘py3-issue’

### CODE REVIEW

File: kobo/xmlrpc.py [hunk 1]
```diff
@@ -489,16 +486,80 @@ def _single_request(self, host, handler, request_body, verbose=0):
             response.read()
         raise xmlrpclib.ProtocolError(host + handler, response.status, response.reason, response.msg)
 
+    def _single_request3(self, host, handler, request_body, verbose=False):
+        # issue XML-RPC request
```
Comment (rohanpm) [inline kobo/xmlrpc.py]: Could you explain a bit, where does this code come from - is it your own invention or was it copy-pasted from somewhere?

Comment (yontalcar) [reply kobo/xmlrpc.py]: It's a copy of `_single_request()` modified to match Py3 `xmlrpc.client.Transport` implementation.

Comment (rohanpm) [reply kobo/xmlrpc.py]: Thanks for explaining.

**Comment by rohanpm** [comment]
This looks fine to me.  I also rebased it and squashed the fixup commit.

The only thing here which might be controversial is that we are locking down some of the log-related methods to dealing with bytes.  It may have been unclear up to now whether it was intended to leave those dealing with bytes, or to make them deal with strings as part of moving to Python 3.  In my opinion, using bytes is correct, given that kobo never enforced any particular encoding on log files, and has no way of knowing how a log file is encoded.

@yontalcar are you fine with this being merged now?

**Comment by yontalcar** [comment]
yeah, sure

**Pull request merged**